src/CommitmentsV2/SFA.DAS.CommitmentsV2/Validators/DraftApprenticeshipDetailsValidator.cs

```diff
@@ -1,34 +1,108 @@
-﻿using System;
-using System.Threading;
-using System.Threading.Tasks;
-using FluentValidation;
+﻿using FluentValidation;
 using FluentValidation.Validators;
 using SFA.DAS.CommitmentsV2.Domain.Entities;
 using SFA.DAS.CommitmentsV2.Domain.Extensions;
 using SFA.DAS.CommitmentsV2.Domain.Interfaces;
 using SFA.DAS.CommitmentsV2.Domain.ValueObjects;
-using SFA.DAS.Reservations.Api.Client;
-using SFA.DAS.Reservations.Api.Client.Types;
 using TrainingProgrammeStatus = SFA.DAS.Apprenticeships.Api.Types.TrainingProgrammeStatus;
 
 namespace SFA.DAS.CommitmentsV2.Validators
 {
     public class DraftApprenticeshipDetailsValidator : AbstractValidator<DraftApprenticeshipDetails>
     {
+        private class StartDateValidator : AbstractValidator<DraftApprenticeshipDetails>
+        {
+            public StartDateValidator(IAcademicYearDateProvider academicYearDateProvider)
+            {
+                RuleFor(ctx => ctx.StartDate)
+                    .GreaterThanOrEqualTo(Constants.DasStartDate)
+                    .WithMessage($"The start date must not be earlier than {Constants.DasStartDate:MM yyyy}");
+
+                RuleFor(ctx => ctx.StartDate)
+                    .Must((draftApprenticeship, startDate) =>
+                        draftApprenticeship.TrainingProgramme.IsActiveOn(draftApprenticeship.StartDate))
+                    .When(ctx => ctx.TrainingProgramme != null)
+                    .WithMessage(draftApprenticeship =>
+                    {
+                        var suffix =
+                            draftApprenticeship.TrainingProgramme.GetStatusOn(draftApprenticeship.StartDate.Value) ==
+                            TrainingProgrammeStatus.Pending
+                                ? $"after {draftApprenticeship.TrainingProgramme.EffectiveFrom.Value.AddMonths(-1):MM yyyy}"
+                                : $"before {draftApprenticeship.TrainingProgramme.EffectiveTo.Value.AddMonths(1):MM yyyy}";
+
+                        return $"This training course is only available to apprentices with a start date {suffix}";
+                    });
+
+                RuleFor(ctx => ctx.StartDate)
+                    .LessThanOrEqualTo(draftApprenticeship =>
+                        academicYearDateProvider.CurrentAcademicYearEndDate.AddYears(1))
+                    .WithMessage(
+                        "The start date must be no later than one year after the end of the current teaching year");
+            }
+        }
+
+        private class EndDateValidator : AbstractValidator<DraftApprenticeshipDetails>
```

Comment:
I like the fact you have separated out the different validation rules, would be nice to be able to test these rules in separate unit tests

Reply:
The rules are tested in the existing test class. These are not public (so should not be unit tested directly) as they are internal helpers to organise the code. We could make the classes public, but I don't see what advantage we get in doing so - the test coverage would be exactly the same same and we'd pollute the namespace (and these non-needed validators would be injected into the DomainValidator).